python_modules/dagster/dagster/components/core/load_defs.py

```diff
@@ -70,6 +71,25 @@ def get_project_root(defs_root: ModuleType) -> Path:
     raise FileNotFoundError("No project root with pyproject.toml or setup.py found")
 
 
+@public
+@preview(emit_runtime_warning=False)
+@suppress_dagster_warnings
+def build_defs_for_component(*, component: Component) -> Definitions:
+    """Constructs Definitions from a standalone component. This is useful for
+    loading individual components in a non-component project.
+
+    Args:
+        component (Component): The component to load defs from.
+    """
+    from dagster.components.core.defs_module import CompositeYamlComponent
+
+    if isinstance(component, CompositeYamlComponent):
```

Comment:
do we need to handle folder and python composite here too ?

Reply:
removed special handling for now, the main path to recommend using here is `Component.from_attributes_dict` or calling the constructor directly I think